addons/website_sale/static/src/js/website_sale.js

```diff
@@ -162,13 +162,33 @@ publicWidget.registry.WebsiteSale = publicWidget.Widget.extend(VariantMixin, {
         this._changeCountry = _.debounce(this._changeCountry.bind(this), 500);
 
         this.isWebsite = true;
+        this.hashProcessed = false;
     },
     /**
      * @override
      */
     start: function () {
+        var self = this;
         var def = this._super.apply(this, arguments);
 
+        var hash = window.location.hash.substring(1);
+        if (hash) {
+            var variantIds = decodeURIComponent(hash).split(',');
+            var $inputs = self.$('input.js_variant_change, select.js_variant_change option');
+            _.each(variantIds, function (id) {
+                var $toSelect = $inputs.filter('[data-value_id="' + id + '"]');
+                if ($toSelect.is('input[type="radio"]')) {
+                    $toSelect.prop('checked', true).attr('checked', 'checked').trigger('change');
```

Comment:
In fact, seeing the code below, there is already a system which will add `attr('checked')` accordingly on variant change right ? Also what about calling `onChangeVariant` instead of triggering the change event ?

Reply:
You are right about the `attr('checked')` being added in `onChangeVariant` I will modify accordingly.
Also I thaught about it but I don't know how to provide the event parameter to `onChangeVariant` if I call it manually. How can I create an event from an element ?

Reply:
We are kinda replying to one another at the same time ^^ the response is in there:
> Indeed, that is why we should not simulate change events :wink: On hash processing, do not trigger `change` but call some methods, and move the relevant code of `onChangeVariant` in those methods.

Reply:
Ok my understanding of the onChangeVariant mechanism was a little light...
I see now that `this.triggerVariantChange(this.$el);` is called in the start method. This will  be sufficient to compute our changes. I than extracted the `_onChangeColorAttribute` method out so that I call it manually to select colors.